homeassistant/components/npi_gpio/binary_sensor.py

```diff
@@ -0,0 +1,89 @@
+"""Support for binary sensor using NPi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import npi_gpio
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BOUNCETIME = 'bouncetime'
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_PORTS = 'ports'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_BOUNCETIME = 50
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_PULL_MODE = 'UP'
+
+
+_SENSORS_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORTS): _SENSORS_SCHEMA,
+    vol.Optional(CONF_BOUNCETIME, default=DEFAULT_BOUNCETIME): cv.positive_int,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    vol.Optional(CONF_PULL_MODE, default=DEFAULT_PULL_MODE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Raspberry PI GPIO devices."""
+    pull_mode = config.get(CONF_PULL_MODE)
+    bouncetime = config.get(CONF_BOUNCETIME)
+    invert_logic = config.get(CONF_INVERT_LOGIC)
+
+    binary_sensors = []
+    ports = config.get('ports')
```

Comment:
Use `CONF_PORTS`.

Please use `dict[key]` for required config keys and keys with default config schema values.

Reply:
> Please use `dict[key]` for required config keys and keys with default config schema values.

...sorry, but i do not undertand well how to apply this.
Could you give me a code example? 
thanks

Reply:
```py
ports = config[CONF_PORTS]
```

Reply:
Done in 8d71d7b485e50f5abf46a3fafd0739c772bbe201 dc14a6aee372fd7a9e87f9af7dcfde1747ccf640 ffe24b2e44277423971addad5d6b10e633091fc5

Reply:
tested and works